RFCs/RFC-0005: Protocol Upgrade to Enable Automatic Distribution of Delegation Rewards.md

```diff
@@ -0,0 +1,49 @@
+# RFC-0005: [Protocol Upgrade to Enable Automatic Distribution of Delegation Rewards]
+
+- **Intent**: [To request a protocol upgrade for automatic distribution of delegation rewards.]
+- **Submitted by**: [lamps, discord: lamps6718, twitter:lamps945 ]
+- **Submitted on**: [26 Dec 2023]
+
+## Abstract
+
+This RFC aims to address one big pain point affecting all node operators on Mina network, which is the distribution of delegation rewards. The current mechanism relies the node operators to voluntarily calculate the rewards and send them back to the delegators. It is error-prone for the calculations and, more importantly, relies complete trust of the node operator, which hinders the decentralisation of the network. The RFC requests a redesign of the mechanism and achieve one that is similar to the Cosmos chains of Cardano, where the rewards are not accessible to the node operators, and the delegators can check and claim the rewards any time.   
+
+## Introduction
+
+A big pain point for all node operators on Mina is how delegation rewards are distributed. Currently, all block rewards go to operators' wallet, and it relies on the operators to calculate the amounts and send them to the delegators. The calcualtion is not trivial and can lead to mistakes, especially when one receives delegations from both locked and unlocked accounts and only the latter is eligible for supercharged rewards (to be removed in hardfork). Moreover, this entire process is voluntary and there has been one or two cases right after mainnet launch in 2021, where the operator ran away with the foundation delegation rewards, and nobody could do anything about it. 
+
+Two major issues come with this relatively rudimentary design: one is the congestions of the network at the end of each epoch when the snapshot of ledger is taken, which resulting in most operators choosing to send transaction during such times. The other is that the Mina holders will only delegate to the operators who they can completely trust, to avoid risking loss of rewards, which hurts the decentralisation of the protocol and results in the funds being concentrated to a small number of reputable operators, instead of having a long tail distribution of decent-sized operators and enthusiasts.
+
+## Objectives
+
+A redesign of the mechanism and achieve one that is similar to the Cosmos chains of Cardano, where the rewards are not accessible to the node operators, and the delegators can check and claim the rewards any time. 
+
+This will help resolve both issues mentioned above: the reward claim transactions can be much evenly distributed in time during each epoch (instead of congestion towards the end/beginning), and the necessity to trust a node operator is removed.  
```

Comment:
Another objective could be to remove as much system complexity as possible from the end-user's experience, in this case, a user staking MINA in a wallet. In my opinion, that user should not be concerned with a "claim" transaction at all; rewards should always be spendable and can be used for things like paying transaction fees for example (reducing two transactions into one -- one claim, one spend -> one spend).

A scenario where a wallet user would need to be aware of a "claim" transaction is if there were constraints around the distribution of rewards, notably, this was a large topic in 2023 in the Cardano ecosystem; [contingent staking](https://forum.cardano.org/t/personal-thoughts-on-contingent-staking/114912). I wonder if there is a case where the Mina ecosystem believes a requirement for this feature would be to enable constraints on the spending (i.e. claiming) of staking rewards, in essence, to give those accounts programmable logic.